example/particles/particles.cpp

```diff
@@ -259,43 +277,27 @@ TaskStatus TransportParticles(MeshBlock *pmb, StagedIntegrator *integrator) {
 
   auto swarm_d = swarm->GetDeviceContext();
 
-  ParArrayND<Real> t("time", max_active_index + 1);
-
-  // Simple particle push: push particles half a zone width until they have
+  // Simple particle push: push particles half the smallest zone width until they have
   // traveled one integrator timestep's worth of time
   pmb->par_for(
       "TransportParticles", 0, max_active_index, KOKKOS_LAMBDA(const int n) {
         if (swarm_d.IsActive(n)) {
-          t(n) = 0.;
           Real v = sqrt(vx(n) * vx(n) + vy(n) * vy(n) + vz(n) * vz(n));
-          while (t(n) < dt) {
+          while (t(n) < t0 + dt) {
             Real dt_cell = dx_push / v;
-            Real dt_end = dt - t(n);
+            Real dt_end = t0 + dt - t(n);
             Real dt_push = std::min<Real>(dt_cell, dt_end);
 
             x(n) += vx(n) * dt_push;
             y(n) += vy(n) * dt_push;
             z(n) += vz(n) * dt_push;
             t(n) += dt_push;
 
-            // Periodic boundaries
-            if (x(n) < x_min) {
-              x(n) = x_max - (x_min - x(n));
-            }
-            if (x(n) > x_max) {
-              x(n) = x_min + (x(n) - x_max);
-            }
-            if (y(n) < y_min) {
-              y(n) = y_max - (y_min - y(n));
-            }
-            if (y(n) > y_max) {
-              y(n) = y_min + (y(n) - y_max);
-            }
-            if (z(n) < z_min) {
-              z(n) = z_max - (z_min - z(n));
-            }
-            if (z(n) > z_max) {
-              z(n) = z_min + (z(n) - z_max);
+            int neighborBlockIndex = swarm_d.GetNeighborBlockIndex(n, x(n), y(n), z(n));
```

Comment:
You are not using neighborBlockIndex, are you missing logic or should this be removed?

Reply:
Yes this can be removed. The call structure here was a bit of a mess though -- `IsOnCurrentMeshBlock` requires that you've already called `GetNeighborBlockIndex` which isn't necessarily obvious -- I'll just merge these functions

Reply:
Oh, I see, yes the method names should reflect the intent. Might want a method with a name like update to help make the intent more explicit.